### PR TITLE
recover gracefully when openai returns 500 or times out

### PR DESCRIPTION
**Pull request by davidsbailey**
## Depends on
* https://github.com/code-dot-org/code-dot-org/pull/58411

See that PR for details.

## Description

Finishes https://codedotorg.atlassian.net/browse/AITT-464
* when openai returns 500, aiproxy returns 503, with "openai" somewhere in the response text
* when the call to openai times out, aiproxy returns 504, with "openai" somewhere in the response text

## Testing story

- [x] new unit tests
- [x] end-to-end testing in previous PR

## Accuracy

no change to accuracy

## API Changes

API now returns 503 and 504 in some cases

### CODE REVIEW

File: lib/assessment/label.py [hunk 1]
```diff
@@ -246,10 +253,9 @@ def label_student_work(self, prompt, rubric, student_code, student_id, examples=
         # Send data to LLM for assessment
         try:
             ai_result = self.ai_label_student_work(prompt, rubric, student_code, student_id, examples=examples, num_responses=num_responses, temperature=temperature, llm_model=llm_model, response_type=response_type)
-        except requests.exceptions.ReadTimeout:
-            logging.error(f"{student_id} request timed out in {(time.time() - start_time):.0f} seconds.")
-            ai_result = None
-
+        except requests.exceptions.ReadTimeout as exception:
+            logging.info(f"{student_id} request timed out in {(time.time() - start_time):.0f} seconds.")
```
Comment (Nokondi) [inline lib/assessment/label.py:257]: Just curious: why are we switching from error to info level for logging errors?

**Review by Nokondi — COMMENTED** [review]
Looks great! Just had one question about logging.

File: lib/assessment/label.py (hunk 1 above)
Comment (davidsbailey) [reply lib/assessment/label.py:257]: this case is now being handled -- I'd rather save error logging for cases we in an unexpected / error state.

Comment (davidsbailey) [reply lib/assessment/label.py:257]: here is one set of opinions: https://stackoverflow.com/questions/2031163/when-to-use-the-different-log-levels 

log-levels

perhaps warn is better? the process continues by returning an error code to the caller, therefore "warn".

**Pull request merged**